node/src/p2p/callbacks/p2p_callbacks_reducer.rs

```diff
@@ -0,0 +1,384 @@
+                    return;
+                };
+
+                dispatcher.push(
+                    TransitionFrontierSyncLedgerSnarkedAction::PeerQueryAddressError {
+                        peer_id,
+                        rpc_id,
+                        error: PeerLedgerQueryError::Timeout,
+                    },
+                );
+                dispatcher.push(
+                    TransitionFrontierSyncLedgerStagedAction::PartsPeerFetchError {
+                        peer_id,
+                        rpc_id,
+                        error: PeerStagedLedgerPartsFetchError::Timeout,
+                    },
+                );
+                dispatcher.push(TransitionFrontierSyncAction::BlocksPeerQueryError {
+                    peer_id,
+                    rpc_id,
+                    error: PeerBlockFetchError::Timeout,
+                });
+                dispatcher.push(P2pDisconnectionAction::Init {
+                    peer_id,
+                    reason: P2pDisconnectionReason::TransitionFrontierRpcTimeout(rpc_kind),
+                });
+            }
+            P2pCallbacksAction::P2pChannelsRpcResponseReceived {
+                peer_id,
+                id,
+                response,
+            } => {
+                let peer_id = *peer_id;
+                let id = *id;
+
+                match response.as_deref() {
+                    None => {
+                        dispatcher.push(
+                            TransitionFrontierSyncLedgerSnarkedAction::PeerQueryAddressError {
+                                peer_id,
+                                rpc_id: id,
+                                error: PeerLedgerQueryError::DataUnavailable,
+                            },
+                        );
+                        dispatcher.push(
+                            TransitionFrontierSyncLedgerStagedAction::PartsPeerFetchError {
+                                peer_id,
+                                rpc_id: id,
+                                error: PeerStagedLedgerPartsFetchError::DataUnavailable,
+                            },
+                        );
+                        dispatcher.push(TransitionFrontierSyncAction::BlocksPeerQueryError {
+                            peer_id,
+                            rpc_id: id,
+                            error: PeerBlockFetchError::DataUnavailable,
+                        });
+                    }
+                    Some(P2pRpcResponse::BestTipWithProof(resp)) => {
+                        let (body_hashes, root_block) = &resp.proof;
+
+                        let (Ok(best_tip), Ok(root_block)) = (
+                            BlockWithHash::try_new(resp.best_tip.clone()),
+                            BlockWithHash::try_new(root_block.clone()),
+                        ) else {
+                            openmina_core::error!(meta.time(); "P2pRpcResponse::BestTipWithProof: invalid blocks");
```

Comment:
this can be replaced by `bug_condition!` right?

Reply:
Nevermind, this is not a case we want to panic when the flag is enabled, it is just that we got bad data in the response.